spec/collator.html

```diff
@@ -73,23 +76,22 @@
           1. If the string given in the Type column of the row is *"boolean"* and value is not *undefined*, then
             1. Let _value_ be ! ToString(_value_).
           1. Set _opt_.[[&lt;_key_&gt;]] to _value_.
-        1. Let _relevantExtensionKeys_ be %Collator%.[[relevantExtensionKeys]].
-        1. Let _r_ be ResolveLocale(%Collator%.[[availableLocales]], _requestedLocales_, _opt_, _relevantExtensionKeys_, _localeData_).
-        1. Set _collator_.[[locale]] to _r_.[[locale]].
+        1. Let _relevantExtensionKeys_ be %Collator%.[[RelevantExtensionKeys]].
+        1. Let _r_ be ResolveLocale(%Collator%.[[AvailableLocales]], _requestedLocales_, _opt_, _relevantExtensionKeys_, _localeData_).
+        1. Set _collator_.[[Locale]] to _r_.[[locale]].
         1. Let _k_ be 0.
         1. Let _rExtensionKeys_ be CreateArrayFromList(_relevantExtensionKeys_).
         1. Let _len_ be ! ToLength(! Get(_rExtensionKeys_, *"length"*)).
         1. Repeat while _k_ < _len_:
           1. Let _key_ be ! Get(_rExtensionKeys_, ! ToString(_k_)).
           1. If _key_ is *"co"*, then
-            1. Let _property_ be *"collation"*.
             1. Let _value_ be _r_.[[co]].
             1. If _value_ is *null*, let _value_ be *"default"*.
+            1. Set _collator_.[[Collation]] to _value_.
           1. Else use the row of <emu-xref href="#table-collator-options"></emu-xref> that contains _key_ in the Key column:
-            1. Let _property_ be the name given in the Property column of the row.
             1. Let _value_ be _r_.[[&lt;_key_&gt;]].
             1. If the name given in the Type column of the row is *"boolean"*, let _value_ be the result of comparing value with *"true"*.
-          1. Set _collator_.[[&lt;_property_&gt;]] to _value_.
```

Comment:
It took some time, and some big refactors, but we are getting there :). Still using it in a couple of places where a record with fields is created and matching some options keys, but it is at least narrowed down to one or two abstracts at the moment.